src/python_testing/TC_EGCTestBase.py

```diff
@@ -0,0 +1,81 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing import matter_asserts
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class ElectricalGridConditionsTestBaseHelper:
+
+    # Spec derived constants
+    kMaxForecastEntries = 56    # Maximum number of list entries for Forecasts
+
+    # Test event trigger IDs
+    kEventTriggerCurrentConditionsUpdate = 0x00A0000000000000
+    kEventTriggerForecastConditionsUpdate = 0x00A0000000000001
+
+    def check_ForecastConditions(self,
+                                 cluster: Clusters.ElectricalGridConditions = Clusters.ElectricalGridConditions,
+                                 forecastConditions: list = None):
+        matter_asserts.assert_list(forecastConditions, "ForecastConditions must be a list")
+        matter_asserts.assert_list_element_type(
+            forecastConditions, cluster.Structs.ElectricalGridConditionsStruct,
+            "ForecastConditions attribute must contain ElectricalGridConditionsStruct elements",
+            allow_empty=True)
+
+        asserts.assert_less_equal(len(forecastConditions),
+                                  self.kMaxForecastEntries,
+                                  f"ForecastConditions list must be less than {self.kMaxForecastEntries} entries")
+        for item in forecastConditions:
+            self.check_ElectricalGridConditionsStruct(cluster=cluster, struct=item)
+
+    def check_ElectricalGridConditionsStruct(self,
+                                             cluster: Clusters.ElectricalGridConditions = Clusters.ElectricalGridConditions,
+                                             struct: Clusters.ElectricalGridConditions.Structs.ElectricalGridConditionsStruct = None):
```

Comment:
```suggestion
                                             cluster: Clusters.ElectricalGridConditions,
                                             struct: Clusters.ElectricalGridConditions.Structs.ElectricalGridConditionsStruct):
```

Same here: we always set these, otherwise typing is wrong (None must be an Optional type)

Reply:
Done

Reply:
Done